packages/components/src/entities/SampleListingPage.tsx

```diff
@@ -0,0 +1,549 @@
+        <>
+            Background import of {nounPlural}
+            {fromFile} completed. To work with the imported samples,&nbsp;
+            <a
+                onClick={() =>
+                    selectGridIdsFromTransactionId(
+                        sampleListingGridId,
+                        SchemaQuery.create(SCHEMAS.SAMPLE_SETS.SCHEMA, sampleType),
+                        transactionAuditId,
+                        SAMPLES_KEY,
+                        actions
+                    )
+                }
+            >
+                select them in the grid.
+            </a>
+        </>
+    );
+};
+
+export const getSamplesCreatedSuccessMessage = (
+    sampleType: string,
+    transactionAuditId: number,
+    createdSampleCount: number,
+    importedSampleCount: number,
+    showAddToStorage: boolean,
+    sampleListingGridId: string,
+    navigate: (url: string | AppURL) => void,
+    actions: Actions,
+    nounSingular = 'item',
+    nounPlural = 'items'
+): ReactNode => {
+    const count = createdSampleCount > 0 ? createdSampleCount : importedSampleCount;
+    const noun = count == 1 ? ' ' + nounSingular : ' ' + nounPlural;
+    const itThem = count == 1 ? 'it' : 'them';
+    const action = createdSampleCount > 0 ? 'created' : 'imported';
+
+    return (
+        <>
+            Successfully {action} {count} {noun}.&nbsp;
+            {showAddToStorage && (
+                <>
+                    <a
+                        onClick={() =>
+                            selectSamplesAndAddToStorage(
+                                sampleType,
+                                count,
+                                transactionAuditId,
+                                sampleListingGridId,
+                                navigate,
+                                actions
+                            )
+                        }
+                    >
+                        Add {itThem} to storage
+                    </a>
+                    &nbsp;now or
+                </>
+            )}
+            {transactionAuditId && (
+                <>
+                    <a
+                        onClick={() =>
+                            selectGridIdsFromTransactionId(
+                                sampleListingGridId,
+                                SchemaQuery.create(SCHEMAS.SAMPLE_SETS.SCHEMA, sampleType),
+                                transactionAuditId,
+                                SAMPLES_KEY,
+                                actions
+                            )
+                        }
+                    >
+                        &nbsp;{showAddToStorage ? 'select' : 'Select'} {itThem} in the grid&nbsp;
+                    </a>
+                </>
+            )}
+            to work with {itThem}.
+        </>
+    );
+};
+
+interface OwnProps {
+    menu: ProductMenuModel;
+    menuInit: (invalidate?: boolean) => void;
+    navigate: (url: string | AppURL, replace?: boolean) => void;
```

Comment:
I'm wondering if we shouldn't have a CommonPageProps or some such thing since we're adding these props in many places as we port page bodies into ui-components.

Reply:
yeah probably. Looks like there are only 3 or 4 such page components at the moment that take these props. I'll go ahead and make that refactor now.